src/System.Console/tests/RedirectedStream.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using Xunit;
+
+public class RedirectedStream : RemoteExecutorTestBase
+{
+    [Fact] // the CI system redirects stdout, so we can only really test the redirected behavior.
```

Comment:
That got me curious as well and I am looking at it in parallel. However, my guess is that this happens because the parent process handles will also be inherited and so, if the parent process is redirected it affects the child. Note that if I run this explicitly by calling
CoreRun.exe xunit.console.netcore.exe System.Console.Tests.dll  -xml testResults.xml -notrait category=nonwindowstests  -notrait category=OuterLoop -notrait category=failing
things work as expected.